providers/targeted_refresh.md

```diff
@@ -0,0 +1,180 @@
+## Targeted Refresh
+
+If you followed the [Creating a new provider](writing_a_new_provider.md) guide then you already have a provider with full refresh working.  Or maybe you're working on an existing provider that only has full refresh and you'd like to add targeted refresh to it.
+
+### What is targeted refresh?
+
+First we should get some technicalities out of the way.  All refreshes have a target, "full" refresh is when the target is the top-level EMS and "targeted" refresh is when the target is anything else (for example a single VM).
+
+The primary benefit of targeted refresh is speed.
+
+Even a moderately large provider will start to take minutes to collect and save all of the inventory.  Doing this simply to pick up a single new VM is extremely wasteful.  Also since only one refresh is able to run at a time due to concurrency concerns, if a full refresh takes 10 minutes and you queue up your refresh request right after it starts the "latency" to getting your change into the database could take almost 20 minutes.  By contrast fetching a single resource like a VM and related inventory is extremely fast.
```

Comment:
```suggestion
Even a moderately large provider will start to take minutes to collect and save all of the inventory.  Doing this simply to pick up a single new VM is extremely wasteful.  Also since only one refresh is able to run at a time due to concurrency concerns, if a full refresh takes 10 minutes and you queue up your refresh request right after it starts, the "latency" to getting your change into the database could take almost 20 minutes.  By contrast fetching a single resource like a VM and related inventory is extremely fast.
```

Reply:
For some reason I couldn't read this sentence without that one comma :)

Reply:
Bit of a run-on already, I'll split it up into two sentences.

Reply:
:+1: done